### PR TITLE
Small update to scipy FFT tutorial

### PR DESCRIPTION
**Pull request by mreineck**
Apart from removing a note in the `scipy` FFT tutorial which is probably no longer necessary, this updates the docs to mention that `scipy.fft` has support for DCT/DST type IV.

The documentation for DCTs and DSTs seems a bit inconsistent, so I wasn't entirely sure which one I should adopt. Maybe some more extensive rework would be useful here, but unfortunately I don't have the time.

### CODE REVIEW

**Comment by mreineck** [comment]
CI errors seem unrelated.

**Review by peterbell10 — CHANGES_REQUESTED** [review]

File: doc/source/tutorial/fft.rst [hunk 1]
```diff
@@ -342,6 +334,25 @@ or, for ``norm='ortho'``:
...
 
+Type IV DCT
+___________
+
+SciPy uses the following definition of the unnormalized DCT-IV
+(``norm=None``):
+
+.. math::
+
+    y[k] = \sum_{n=0}^{N-1} x[n] \cos\left({\pi (2n+1)(2k+1) \over 2N}\right)
```
[the remark this answers is not in the document]
Comment (mreineck) [reply doc/source/tutorial/fft.rst]: Sorry, that's a mixup of two independent errors ... should be fixed now.
I also changed 2N to 4N in the normalized expression, the factor in front of the sum should be OK though.

File: doc/source/tutorial/fft.rst [hunk 2]
```diff
@@ -486,6 +497,24 @@ the following definition of the unnormalized DST-III (``norm=None``):
...
 
+Type IV DST
+___________
+
+SciPy uses the following definition of the unnormalized DST-IV
+(``norm=None``):
+
+.. math::
+
+    y[k] = \sum_{n=0}^{N-1} x[n] \sin\left({\pi (2n+1)(2k+1) \over 2N}\right)
```
[the remark this answers is not in the document]
Comment (mreineck) [reply doc/source/tutorial/fft.rst]: Fixed, see above.

**Review by peterbell10 — APPROVED** [review]

**Comment by peterbell10** [comment]
CI failures unrelated again so merged. Thanks @mreineck.

**Comment by hgomersall** [comment]
I appreciate I'm rather late to the boat on this, but I'm curious as to the reason why the link to `pyfftw` was removed (though the link _was_ to the wrong site)?

**Comment by ilayn** [comment]
Probably because we started using pocketFFT covered the speed part good enough and also not used in SciPy so no point on disturbing the tutorial flow.

**Pull request merged**